ide/static/ide/js/resources.js

```diff
@@ -555,68 +588,76 @@ CloudPebble.Resources = (function() {
             }
 
             var update_font_preview = function(group) {
-                group.find('.font-preview').remove();
-                var regex_str = group.find('.edit-resource-regex').val();
-                var id_str = group.find('.edit-resource-id').val();
-                var preview_regex = new RegExp('');
-                try {
-                    preview_regex = new RegExp(regex_str ? regex_str : '.', 'g');
-                    group.find('.font-resource-regex-group').removeClass('error').find('.help-block').text(gettext("A PCRE regular expression that restricts characters."));
-                } catch(e) {
-                    group.find('.font-resource-regex-group').addClass('error').find('.help-block').text(e);
-                }
-                var tracking = parseInt(group.find('.edit-resource-tracking').val(), 10) || 0;
-
-                _.each(resource.variants, function(tags) {
-                    var row = $('<div class="control-group font-preview"><label class="control-label">' + gettext('Preview') + '</label>');
-                    var preview_holder = $('<div class="controls">');
-                    $('<div class="font-tag-preview">').appendTo(preview_holder).text(_.chain(tags).map(get_tag_data_for_id).pluck('name').value().join(', ') || gettext("No tags"));
-                    var preview = $('<div>').appendTo(preview_holder);
-                    var line = ('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789~!@#$%^& *()_+[]{}\\|;:\'"<>?`'.match(preview_regex)||[]).join('');
-                    var font_size = id_str.match(/[0-9]+$/)[0];
-
-                    preview.text(line);
-                    // Officially, a CSS pixel is defined as one pixel at 96 dpi.
-                    // 96 / PEBBLE_PPI should thus be correct.
-                    // We use 'transform' to work around https://bugs.webkit.org/show_bug.cgi?id=20606
-                    preview.css({
-                        'font-family': CloudPebble.Resources.GetFontFamily(resource, tags),
-                        'font-size': font_size + 'px',
-                        'line-height': font_size + 'px',
-                        'letter-spacing': tracking + 'px',
-                        'transform': 'scale(' + (96 / PEBBLE_PPI) + ')',
-                        'transform-origin': '0 0',
-                        'display': 'inline-block',
-                        'border': (2 * (PEBBLE_PPI / 96)) + 'px solid #767676',
-                        'padding': '5px',
-                        'border-radius': '5px',
-                        'background-color': 'white',
-                        'color': 'black'
+                if (resource.kind == 'font') {
```

Comment:
If I do the check for resource.kind == 'font' before the call, it's not! (Which I've just committed)